src/request.ts

```diff
@@ -846,6 +929,10 @@ class DatastoreRequest {
 
     let info: RunQueryInfo;
 
+    if (this.state === TransactionState.EXPIRED) {
+      callback(new Error(transactionExpiredError));
+      return;
+    }
```

Comment:
Do we need to check `readTimeAndConsistencyError` here?

Reply:
No. That check is done further downstream in the call to `runQueryStream`. But I did spot some other cases where errors weren't bubbling up and fixed those bugs.